generators/PHPythia8/PHPythia8.C

```diff
@@ -86,22 +89,36 @@ int PHPythia8::Init(PHCompositeNode *topNode)
   unsigned int seed = PHRandomSeed();
 
   if (seed > 900000000)
-  {
-    seed = seed % 900000000;
-  }
+    {
+      seed = seed % 900000000;
+    }
 
   if ((seed > 0) && (seed <= 900000000))
-  {
-    _pythia->readString("Random:setSeed = on");
-    _pythia->readString(str(boost::format("Random:seed = %1%") % seed));
-  }
+    {
+      //_pythia->readString("Random:setSeed = on");
+      //_pythia->readString(str(boost::format("Random:seed = %1%") % seed));
+    
+      ppGen->readString("Random:setSeed = on");
+      ppGen->readString(str(boost::format("Random:seed = %1%") % seed));
+
+      pnGen->readString("Random:setSeed = on");
+      pnGen->readString(str(boost::format("Random:seed = %1%") % seed));
+    }
   else
+    {
+      cout << PHWHERE << " ERROR: seed " << seed << " is not valid" << endl;
+      exit(1);
+    }
+
+  //if(! _legacy_vertexgenerator) _pythia->init();
+  //else
   {
-    cout << PHWHERE << " ERROR: seed " << seed << " is not valid" << endl;
-    exit(1);
-  }
+    ppGen->readString("Beams:idB = 2212");
+    pnGen->readString("Beams:idB = 2112");
```

Comment:
It seems to always overwrite the settings read from the config file.  If so, when `_legacy_vertexgenerator` is false, does PHPythia8 will work properly as it does?

Reply:
In config file, we have been just defining only Beams:idA as  `Beams:idA= 2212` but not Beams:idB. If we don't define it then I suppose the Beam:idB becomes same as Beam:idA. But here, we are defining two different Beams:id to choose from pp vs pn.

Reply:
I see.  I confirmed that `idB` (and also `idA`) are 2212 by default;
https://pythia.org/latest-manual/BeamParameters.html